Sources/XcodeGenKit/SchemeGenerator.swift

```diff
@@ -83,34 +83,56 @@ public class SchemeGenerator {
 
     public func generateScheme(_ scheme: Scheme) throws -> XCScheme {
 
-        func getBuildEntry(_ buildTarget: Scheme.BuildTarget) -> XCScheme.BuildAction.Entry {
+        func getBuildEntry(_ buildTarget: Scheme.BuildTarget) throws -> XCScheme.BuildAction.Entry {
+            let pbxProj: PBXProj
+            let projectFilePath: String
+            switch buildTarget.target.location {
+            case .project(let project):
+                guard let projectReference = self.project.getProjectReference(project) else {
+                    fatalError("Unable to find project reference named \"\(project)\" in project.yml")
+                }
+                pbxProj = try XcodeProj(pathString: projectReference.path).pbxproj
+                projectFilePath = projectReference.path
+            case .local:
+                pbxProj = self.pbxProj
+                projectFilePath = "\(self.project.name).xcodeproj"
```

Comment:
Is this true?

Can you try `project.basepath` to get xcodeproj path.

Reply:
Yes, `ReferencedContainer` is relative to a directory which has `xcodeproj`